core/src/services/hdfs_native/writer.rs

```diff
@@ -15,28 +15,41 @@
 // specific language governing permissions and limitations
 // under the License.
 
+use bytes::{Buf, Bytes};
 use hdfs_native::file::FileWriter;
 
-use crate::raw::oio;
+use crate::raw::*;
+use crate::services::hdfs_native::error::parse_hdfs_error;
 use crate::*;
 
 pub struct HdfsNativeWriter {
-    _f: FileWriter,
+    f: FileWriter,
 }
 
 impl HdfsNativeWriter {
     pub fn new(f: FileWriter) -> Self {
-        HdfsNativeWriter { _f: f }
+        HdfsNativeWriter { f }
     }
 }
 
 impl oio::Write for HdfsNativeWriter {
-    async fn write(&mut self, _bs: Buffer) -> Result<()> {
-        todo!()
+    async fn write(&mut self, mut bs: Buffer) -> Result<()> {
+        while bs.has_remaining() {
+            let n = self
+                .f
+                .write(Bytes::copy_from_slice(bs.chunk()))
```

Comment:
The `Buffer` itself is an iterator of `Bytes`, so we can simplely:

```rust
for chunk in bs {
  self.f.write(chunk).await;
  balabala
}
```

Reply:
Makes sense

Reply:
```
for chunk in bs {
  self.f.write(chunk).await;
  balabala
}
```
Writing like this will report the following error, so the original implementation is still used here
```
`bs` moved due to this implicit call to `.into_iter()`
```